docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,600 @@
+
+.. runrecord:: _examples/DL-101-130-117
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad siblings
+
+.. gitusernote::
+
+   Creating a sibling on GitHub will create a new empty repository under the
+   account that you provide and set up a *remote* to this repository. Upon a
+   :command:`datalad publish` to this sibling, your datasets history
+   will be pushed there.
+
+   .. index:: datalad command; publish
+
+On GitHub, you will see two new, empty repository with the names
+``midtermproject``, and ``inputs``. However, none of these repositories yet contain
+any of your dataset's history or files. This requires *publishing* the current
+state of the dataset to this sibling with the :command:`datalad publish`
+(:manpage:`datalad-publish` manual) command. As before, we do this recursively with
+``-r/--recursive``. Because this will publish the contents of *two* datasets,
+``midterm_project`` and ``inputs``, you will be asked twice to authenticate.
+
+.. code-block:: bash
+
+   $ datalad publish -r --to github
+   [INFO   ] Publishing <Dataset path=/home/me/dl-101/DataLad-101/midterm_project/input> to github
+   publish(ok): input (dataset) [pushed to github: ['[new branch]', '[new branch]']]
+   [INFO   ] Publishing <Dataset path=/home/me/dl-101/DataLad-101/midterm_project> to github
+   publish(ok): . (dataset) [pushed to github: ['[new branch]', '[new branch]']]
+   action summary:
+     publish (ok: 2)
+
+.. gitusernote::
+
+   The :command:`datalad publish` uses ``git push``, and ``git annex copy`` under
+   the hood. Publication targets need to either be configured remote Git repositories,
+   or Git-annex special remotes (if they support data upload).
+
+Here is one important detail, though: By default, your tags will not be published.
+This requires an additional :command:`git push` with the ``--tags`` option to the
+sibling:
+
+.. code-block:: bash
+
+   $ git push github --tags
+   To https://github.com/adswa/midtermproject.git
+     * [new tag]         ready4analysis -> ready4analysis
+
+
+Yay! Consider your midterm project submitted! Others can now install your
+dataset and check out your data science project -- and even better: they can
+reproduce your data science project easily from scratch!
+
+.. findoutmore:: On the looks and feels of this published dataset
+
+   Now that you have created and published such a YODA-compliant dataset, you
+   are understandably excited how this dataset must look and feel for others.
+   Therefore, you decide to install this dataset in a new location on your
+   computer, just to get a feel for it.
+
+   Replace the ``url`` in the :command:`install` command below with the path
+   to your own ``midtermproject`` GitHub repository:
+
+   .. runrecord:: _examples/DL-101-130-119
+      :language: console
+      :workdir: dl-101/DataLad-101/midterm_project
+
+      $ cd ../../
+      $ datalad install -r --source "https://github.com/adswa/midtermproject.git"
```

Comment:
The recursive part is not needed. DataLad does that automatically on `rerun` based on the `input` specification we gave above. I know that this would make it impossible to keep the immediately following paragraphs as they are, but not having to know anything about the specific nature of the dataset in order to rerun IS a killer feature -- that is a bit shadowed by the present approach.